Sources/StreamChat/APIClient/APIClient.swift

```diff
@@ -101,12 +101,34 @@ class APIClient {
         recoveryQueue.addOperation(requestOperation)
     }
 
+    /// Performs a network request and retries in case of network failures. The network operation
+    /// won't be managed by the APIClient instance. Instead it will be added on the provided operationQueue.
+    ///
+    /// - Parameters:
+    ///   - endpoint: The `Endpoint` used to create the network request.
+    ///   - operationQueue: The queue that will be responsible for executing the network operation
+    ///   - completion: Called when the networking request is finished.
+    func unmanagedRequest<Response: Decodable>(
+        endpoint: Endpoint<Response>,
+        operationQueue: OperationQueue,
+        completion: @escaping (Result<Response, Error>) -> Void
+    ) {
+        operationQueue.addOperation(
+            unmanagedOperation(endpoint: endpoint, completion: completion)
+        )
+    }
+
     private func operation<Response: Decodable>(
         endpoint: Endpoint<Response>,
         isRecoveryOperation: Bool,
         completion: @escaping (Result<Response, Error>) -> Void
     ) -> AsyncOperation {
         AsyncOperation(maxRetries: maximumRequestRetries) { [weak self] operation, done in
+            guard self?.isRefreshingToken == false else {
+                completion(.failure(ClientError.RefreshingToken()))
+                return
+            }
```

Comment:
There are 2 problems here:
1. We are not completing the operation (you would need to call `done(.continue)`)
2. We should not be returning this error upwards, because otherwise the whole retry/queueing strategy is lost

Why did you move the logic from `executeRequest` to here?

Reply:
Great point! I updated it